http/client/src/main/java/org/wildfly/security/http/client/ElytronHttpClient.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client;
+
+import org.wildfly.security.http.client.hpi.ClientConfigProvider;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+/**
+ * Http Client side Authentication using java 11 HttpClient
```

Comment:
The HttpClient implementation might have been introduced in java 11 but it is available in later java as well so we should not name it java 11 HttpClient here. Something like `Elytron client for HTTP authentication` might describe this class better.